modules/acm_customer/src/Plugin/rest/resource/CustomerDeleteResource.php

```diff
@@ -24,6 +28,61 @@
  */
 class CustomerDeleteResource extends ResourceBase {
 
+  /**
+   * Current request.
+   *
+   * @var \Symfony\Component\HttpFoundation\Request
+   */
+  private $currentRequest;
+
+  /**
+   * Construct.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param array $serializer_formats
+   *   The available serialization formats.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   * @param \Symfony\Component\HttpFoundation\Request $current_request
+   *   The current request.
+   */
+  public function __construct(
+    array $configuration,
```

Comment:
I think constructor is not required as we not using any service (default logger can be utilized)

Reply:
I'm happy for you to suggest code improvements because I don't understand the inner workings of Drupal's RestResource. What I do know is these functions seem to be in every other RestResource and without them here we get `$data = []` even when request POST body is `{"email":"me@you.com"}`

Reply:
`create()` method is used when we try to override the constructor which is not required in this case. For example see [DBLogResource](https://api.drupal.org/api/drupal/core%21modules%21dblog%21src%21Plugin%21rest%21resource%21DBLogResource.php/class/DBLogResource/8.6.x)

However, regarding empty `$data`, I think right way to confirm this is by creating a test for this (will try to create one) resource because to me it looks fine. Regarding code fix, I think `empty($data['mail'])` check should be enough.

Reply:
plugin for POST (as earlier was get) just for reference [WebformSubmitResource](https://cgit.drupalcode.org/webform_rest/tree/src/Plugin/rest/resource/WebformSubmitResource.php)

Reply:
I agree the constructor override is not obviously required. My concern now is: Why does this `function post()` only work when written this way?